#OpenXT:Seen BSSIDs are populated in database for the purpose of saving them on boot.
#       Reloading of these seen-bssids is done by network-slave on start up.
--- a/src/settings/nm-settings-connection.c
+++ b/src/settings/nm-settings-connection.c
@@ -24,6 +24,8 @@
 #include "nm-settings-connection.h"
 
 #include <string.h>
+#include <errno.h>
+#include <sys/wait.h>
 
 #include "nm-common-macros.h"
 #include "nm-dbus-interface.h"
@@ -64,6 +66,8 @@
 
 static void nm_settings_connection_connection_interface_init (NMConnectionInterface *iface);
 
+static void update_seen_bssids_db ();
+
 G_DEFINE_TYPE_WITH_CODE (NMSettingsConnection, nm_settings_connection, NM_TYPE_EXPORTED_OBJECT,
                          G_IMPLEMENT_INTERFACE (NM_TYPE_CONNECTION, nm_settings_connection_connection_interface_init)
                          )
@@ -718,6 +722,8 @@ remove_entry_from_db (NMSettingsConnecti
 		if (data) {
 			g_file_set_contents (db_file, data, len, &error);
 			g_free (data);
+			if (!strcmp (db_name, "seen-bssids"))
+				update_seen_bssids_db ();
 		}
 		if (error) {
 			_LOGW ("error writing %s file '%s': %s", db_name, db_file, error->message);
@@ -2314,6 +2320,26 @@ nm_settings_connection_has_seen_bssid (N
 	return !!g_hash_table_lookup (NM_SETTINGS_CONNECTION_GET_PRIVATE (self)->seen_bssids, bssid);
 }
 
+static void
+update_seen_bssids_db (void)
+{
+       char *argv[6] = { "/usr/bin/nm_sync.sh", "-n", "seen-bssids", "-w", SETTINGS_SEEN_BSSIDS_FILE, NULL };
+       int err, stat_loc;
+
+       err = fork();
+       if (err == -1) {
+               g_warning("%s.%d: cannot fork to nm_sync.sh: %d", __FILE__, __LINE__, errno);
+       } else if (err) {
+               while ((waitpid(err, &stat_loc, 0) == -1) && (errno == EINTR))
+                       ;
+               err = WEXITSTATUS(stat_loc);
+               if (err)
+                       g_warning("%s.%d: nm_sync.sh failed with code: %d", __FILE__, __LINE__, err);
+       } else {
+               _exit(execvp(argv[0], argv));
+       }
+}
+
 /**
  * nm_settings_connection_add_seen_bssid:
  * @self: the #NMSettingsConnection
@@ -2370,6 +2396,7 @@ nm_settings_connection_add_seen_bssid (N
 	data = g_key_file_to_data (seen_bssids_file, &len, &error);
 	if (data) {
 		g_file_set_contents (SETTINGS_SEEN_BSSIDS_FILE, data, len, &error);
+		update_seen_bssids_db ();
 		g_free (data);
 	}
 	g_key_file_free (seen_bssids_file);
